spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/sql/TestAggregatePushDown.java

```diff
@@ -249,6 +250,78 @@ public void testAggregateNotPushDownIfOneCantPushDown() {
     assertEquals("expected and actual should equal", expected, actual);
   }
 
+  @Test
+  public void testAggregationPushdownStructInteger() {
```

Comment:
an alternative to having multiple methods that effectively call the same test code is to have a single test method that is parameterized

Reply:
I did some more refactoring, while we could use `@Parameterized` in this case it ends up not being super readable compared to just being explicit imo. I've updated the PR so that there's helpers for asserting the aggregates and the expected contents of the explain plan.